docs/HACKING.md

```diff
@@ -1,104 +1,151 @@
+export KUBE_SSH_KEY_PATH=<path to pem>
 ```
 
-### Deploying the operator directly
+#### Set Image Name
+
+This is the name of the operator image that will be pushed to the container registry.
+
 ```shell script
-make deploy IMG=$OPERATOR_IMAGE
+export OPERATOR_IMAGE=quay.io/<quay username>/wmco:<identifying tag>
 ```
-The above command will directly deploy all the resources required for the operator to run, as well as the WMCO pod.
-To remove the installed resources:
+
+#### Set CI to false
+
+This is to tell the tests that this test isn't running in CI.
+
 ```shell script
-make undeploy
+export OPENSHIFT_CI=false
 ```
 
-### Deploying the operator through OLM
+#### Set shared credentials file
+
+If you are running on AWS or GCP, set your shared credentials file.
+
 ```shell script
-hack/olm.sh run -k "<PRIVATE_KEY.PEM>"
+export AWS_SHARED_CREDENTIALS_FILE=<path to aws credentials file>
 ```
-Where `"<PRIVATE_KEY.PEM>"` is the private key which will be used to configure Windows nodes.
-This command builds the operator image, pushes it to remote repository and uses OLM to launch the operator. Executing
-the [Build](#build) step is not required.
 
-In order to build the operator ignoring the existing build image cache, run the above command with the `-i` option.
+```shell script 
+export GOOGLE_CREDENTIALS=<path to google credentials file>
+```
+
+#### Set KUBECONFIG
+
+To run tests, you should have a running cluster, and your KUBECONFIG set.
 
-To clean-up the installation, use:
 ```shell script
-hack/olm.sh cleanup
+export KUBECONFIG=<path to kubeconfig> 
 ```
 
-### Running e2e tests on a cluster
-The following environment variables must be set:
+The operator can now be deployed
+[Manually](#deploying-the-operator-manually),
+[Using the OLM](#running-the-olm-hack-script),
+[Using the e2e tests](#running-the-e2e-tests),
+[Using the e2e tests on platform agnostic infrastructure](#running-e2e-tests-on-platform-agnostic-infrastructure), or
+[Using the official deployment method](#deploying-Using-the-official-deployment-method).
+
+### Deploying the operator manually
+
 ```shell script
-# SSH key to be used for creation of cloud-private-key secret
-export KUBE_SSH_KEY_PATH=<path to RSA type ssh key>
-export OPERATOR_IMAGE=quay.io/<insert username>/wmco:$VERSION_TAG
-export OPENSHIFT_CI=false
-# on AWS only:
-export AWS_SHARED_CREDENTIALS_FILE=<path to aws credentials file>
+make deploy IMG=$OPERATOR_IMAGE
 ```
-Once the above variables are set, run the following script:
+
+The above command will directly deploy all the resources required for the operator to run, as well as the WMCO pod.
+
+### Running the OLM hack script
+
+Deploy the operator lifecycle manager onto your cluster with the hack script.
+
+This command builds the operator image, pushes it to remote repository and uses OLM to launch the operator.
+
 ```shell script
-hack/run-ci-e2e-test.sh
+hack/olm.sh run -k KUBE_SSH_KEY_PATH
+```
+
+### Running the e2e tests
```

Comment:
Why is this under deploy? I understand this also deploys the operator but this heeds a separate header.